Neos.Neos/Classes/Service/Mapping/NodePropertyConverterService.php

```diff
@@ -155,15 +157,30 @@ public function getPropertiesArray(NodeInterface $node)
     }
 
     /**
-     * Convert the given value to a simple type or an array of simple types.
+     * Convert the given value to a simple type or an array of simple types or to a \JsonSerializable.
      *
-     * @param mixed $propertyValue
-     * @param string $dataType
-     * @return mixed
+     * @param mixed $propertyValue the deserialized node property value
+     * @param string $dataType the property type from the node type schema
+     * @return \JsonSerializable|int|float|string|bool|null|array<mixed>
      * @throws PropertyException
      */
     protected function convertValue($propertyValue, $dataType)
     {
+        if ($propertyValue instanceof \JsonSerializable) {
```

Comment:
This is probably breaking for any property that is already JsonSerializable. I am thinking DateTime here for example that would suddenly be stored differntly.

Reply:
As long as datelime cannot de deserialized via fromArray it should not interfere

Reply:
Oh I see, nested if, meh, I guess ` DenormalizingObjectConverter::isDenormalizable($propertyValue::class)` should be the primary check LOL I guess it's true though, that won't interfere then,

Reply:
Ha yes sorry pushed that thing really oft around till it got into a nested if.
Will adjust this but functionality will stay the same. Would please approve this in that case? ^^